serde_arrow/src/internal/deserialization/deserializer.rs

```diff
@@ -0,0 +1,168 @@
+use serde::de::Visitor;
+
+use crate::internal::{
+    deserialization::outer_sequence_deserializer::OuterSequenceDeserializer,
+    error::{fail, Error, Result},
+};
+
+/// A structure to deserialize Arrow arrays into Rust objects
+///
+/// It can be constructed via
+///
+#[cfg_attr(has_arrow, doc = r"- [`Deserializer::from_record_batch`]")]
+#[cfg_attr(has_arrow, doc = r"- [`Deserializer::from_arrow`]")]
+#[cfg_attr(has_arrow2, doc = r"- [`Deserializer::from_arrow2`]")]
+///
+pub struct Deserializer<'de>(pub(crate) OuterSequenceDeserializer<'de>);
```

Comment:
Should this struct still live in `internal`, given that it's now part of the public API?

Reply:
TBH. Also not happy with that. In particular as it is quite deep in the file tree. There is however precedence with the schema functionality. I will move it to the top level of the internal crate as a compromise :)